### ISSUE TITLE
Panic on date_bin with interval 0 

### ISSUE THREAD

**Issue by NGA-TRAN** [apache/arrow-datafusion#6517]
### Describe the bug

If the interval of the `date_bin` function is 0, the query will hit panic 

### To Reproduce

Run the queries below in DataFusion CLI
```SQL
DataFusion CLI v25.0.0
❯ SELECT DATE_BIN(INTERVAL '0 second', TIMESTAMP '2022-08-03 14:38:50Z');
thread 'main' panicked at 'attempt to calculate the remainder with a divisor of zero',  /arrow-datafusion/datafusion/physical-expr/src/datetime_expressions.rs:354:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

❯ SELECT DATE_BIN(INTERVAL '0 day', TIMESTAMP '2022-08-03 14:38:50Z');
thread 'main' panicked at 'attempt to calculate the remainder with a divisor of zero', /arrow-datafusion/datafusion/physical-expr/src/datetime_expressions.rs:354:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

SELECT
  DATE_BIN(INTERVAL '0' minute, time) AS time,
  val
FROM (
  VALUES
    (TIMESTAMP '2021-06-10 17:05:00Z', 0.5),
    (TIMESTAMP '2021-06-10 17:19:10Z', 0.3)
  ) as t (time, val);
thread 'tokio-runtime-worker' panicked at 'attempt to calculate the remainder with a divisor of zero', /arrow-datafusion/datafusion/physical-expr/src/datetime_expressions.rs:354:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
0 rows in set. Query took 0.012 seconds.

```

### Expected behavior

A normal error message rather than a panic. Something like "interval 0 is not supported in date_bin`

**Comment by NGA-TRAN** [comment]
@alamb I am working on this

**Comment by NGA-TRAN** [comment]
Close this as the fixed #6522  is merged
